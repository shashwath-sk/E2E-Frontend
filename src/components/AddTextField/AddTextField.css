.addContentType{
    border: 3px dotted rgb(100,83,209);
    border-radius: 10px;
    margin-top: 15px;
    display: flex;
    justify-content: center;
    height: 50px;
    width: 283px;
    align-items: center;
    padding-left: 10px;
    padding-right: 10px;
    background-color: white;
}
.addContentType>*{
    color:rgb(80,31,198);
    font-weight:500
}
.contents{
    /* justify-content: space-between; */
    border: 3px dotted white;
    border-radius: 10px;
    margin-top: 15px;
    display: flex;
    justify-content: space-between;
    height: 50px;
    width: 283px;
    align-items: center;
    padding-left: 10px;
    padding-right: 10px;
    background-color: white;
}
.contents>*{
    color: rgb(144,144,144);
}

.addTextField{
    border: 3px dotted rgb(100,83,209);
    border-radius: 10px;
    margin-top: 15px;
    display: flex;
    justify-content: center;
    height: 50px;
    width: 44vw;
    align-items: center;
    margin-top: 30px;
    padding-left: 10px;
    padding-right: 10px;
    background-color: white;
}
.addTextField>*{
    color:rgb(80,31,198);
    font-weight:500
}