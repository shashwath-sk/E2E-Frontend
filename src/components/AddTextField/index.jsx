import * as React from 'react';
import PropTypes from 'prop-types';
import './AddTextField.css';

export default function AddTextField( { style,placeholder,placeholder2,contentId,setShowModal,setEntity,handleClick } ) {
    console.log(placeholder,style,placeholder2);
    const handleClickDynamically=()=>{
        if(handleClick)
        {
            handleClick(contentId,'version');
        }
        else
        {
            setEntity(true);
            setShowModal(true);
        }
    };
    return(
        <div className={`${style}`} onClick={()=>handleClickDynamically()}>
            <h3>{placeholder}</h3>
            {placeholder2&&<h3>{placeholder2}</h3>}

        </div>
    );

}


AddTextField.propTypes = {
    placeholder: PropTypes.string.isRequired,
    style: PropTypes.string.isRequired,
    placeholder2: PropTypes.string.isRequired,
    contentId: PropTypes.string.isRequired,
    handleClick: PropTypes.func.isRequired,
    setShowModal: PropTypes.func.isRequired,
    setEntity: PropTypes.func.isRequired,
};