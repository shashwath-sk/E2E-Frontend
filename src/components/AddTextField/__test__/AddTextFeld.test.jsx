import AddTextField from '../';
import React from 'react';
import { render, screen } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';

describe('AddTextField', () => {
    it('should render AddTextField', () => {
        const { asFragment } = render(
            <BrowserRouter>
                <AddTextField placeholder="Placeholder" />
            </BrowserRouter>
        );
        expect(asFragment).toMatchSnapshot();
    });

    it('should display placeholder text when component is rendered with placeholder text', () => {
        render(
            <BrowserRouter>
                <AddTextField placeholder="Placeholder" />
            </BrowserRouter>
        );
        expect(screen.getByText('Placeholder')).toBeInTheDocument();
    }
    );

});