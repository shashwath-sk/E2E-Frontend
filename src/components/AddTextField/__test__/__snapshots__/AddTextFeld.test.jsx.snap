// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddTextField should render AddTextField 1`] = `[Function]`;
